File: python-sdk/src/cosevi_open_data/known_dashboards.py
"""This file stores known COSEVI dashboard metadata for the Python SDK."""

from __future__ import annotations

from dataclasses import dataclass

# -----------------------------------------------------------------------------
# Dashboard dataclass
# -----------------------------------------------------------------------------


# -----------------------------------------------------------------------------
# Dashboard metadata
# -----------------------------------------------------------------------------
@dataclass
class KnownCoseviDashboard:
    """A known COSEVI dashboard entry with a stable key and Junar GUID."""

    key: str
    """Stable snake_case key for use in code."""

    guid: str
    """Junar GUID used as the URL path segment when calling the API."""

    title: str
    """Human-readable title in Spanish."""

    category: str
    """Category for filtering such as fallecidos or accidentes."""

    description: str | None = None
    """Optional description of the dashboard content."""


# -----------------------------------------------------------------------------
# Dashboard registry
# -----------------------------------------------------------------------------

KNOWN_COSEVI_DASHBOARDS: list[KnownCoseviDashboard] = [
    KnownCoseviDashboard("datos_pais", "DATOS-PAIS", "Informaci?n general de Costa Rica", "general"),
    KnownCoseviDashboard("infracciones", "INFRA-43614", "Infracciones", "infracciones"),
    KnownCoseviDashboard("fallecidos_en_sitio", "FALLE-EN-SITIO", "Fallecidos en sitio", "fallecidos"),
    KnownCoseviDashboard("accidentes_transito", "ACCID-17064", "Accidentes de tr?nsito", "accidentes"),
    KnownCoseviDashboard("conductores_licencias", "ACRED-DE-CONDU-2", "Conductores y licencias", "licencias"),
    KnownCoseviDashboard("pruebas_teoricas_practicas", "PRUEB-TEORI-Y-PRACT", "Pruebas te?ricas y pr?cticas", "pruebas"),
    KnownCoseviDashboard("tabla_interactiva_fallecidos", "DATOS-PARA-TABLA-INTER-94312", "Datos para tabla interactiva de fallecidos", "fallecidos"),
    KnownCoseviDashboard("tabla_interactiva_accidentes", "DATOS-PARA-TABLA-INTER-DE", "Datos para tabla interactiva de accidentes", "accidentes"),
    KnownCoseviDashboard("accidentes_con_victimas", "ACCID", "Accidentes con v?ctimas", "accidentes"),
    KnownCoseviDashboard("caracteristicas_infracciones", "CARAC-DE-INFRA", "Caracter?sticas de infracciones", "infracciones"),
    KnownCoseviDashboard("consulta_infracciones_articulo", "CONSU-DE-INFRA-POR-ARTIC", "Consulta de infracciones por art?culo", "infracciones"),
    KnownCoseviDashboard("infracciones_detalle", "INFRA-46061", "Infracciones detalle", "infracciones"),
]
"""Complete list of known COSEVI dashboards with stable keys and GUIDs."""

# -----------------------------------------------------------------------------
# Lookup helpers
# -----------------------------------------------------------------------------


# Returns the known dashboards, optionally filtered by category.
def list_known_dashboards(category: str | None = None) -> list[KnownCoseviDashboard]:
    """Return all known dashboards, optionally filtered by category."""
    if not category:
        return list(KNOWN_COSEVI_DASHBOARDS)
    return [dashboard_entry for dashboard_entry in KNOWN_COSEVI_DASHBOARDS if dashboard_entry.category == category]


# Returns one known dashboard by stable key or GUID.
def get_known_dashboard(key_or_guid: str) -> KnownCoseviDashboard | None:
    """Look up a known dashboard by its stable key or Junar GUID."""
    for dashboard_entry in KNOWN_COSEVI_DASHBOARDS:
        if dashboard_entry.key == key_or_guid or dashboard_entry.guid == key_or_guid:
            return dashboard_entry
    return None

File: javascript-sdk/src/known-dashboards.ts
// -----------------------------------------------------------------------------
// Types
// -----------------------------------------------------------------------------

/** A known COSEVI dashboard entry with stable key and GUID. */
export interface KnownCoseviDashboard {
  /** Stable camelCase/snake_case key for use in code. */
  key: string;
  /** Junar GUID used as the URL path segment when calling the API. */
  guid: string;
  /** Human-readable title in Spanish. */
  title: string;
  /** Category for filtering (e.g., "fallecidos", "accidentes"). */
  category: string;
  /** Optional description of the dashboard content. */
  description?: string;
}

// -----------------------------------------------------------------------------
// Dashboard registry
// -----------------------------------------------------------------------------

/** Complete list of known COSEVI dashboards with stable keys and GUIDs. */
export const KNOWN_COSEVI_DASHBOARDS: KnownCoseviDashboard[] = [
  { key: "datos_pais", guid: "DATOS-PAIS", title: "Información general de Costa Rica", category: "general" },
  { key: "infracciones", guid: "INFRA-43614", title: "Infracciones", category: "infracciones" },
  { key: "fallecidos_en_sitio", guid: "FALLE-EN-SITIO", title: "Fallecidos en sitio", category: "fallecidos" },
  { key: "accidentes_transito", guid: "ACCID-17064", title: "Accidentes de tránsito", category: "accidentes" },
  { key: "conductores_licencias", guid: "ACRED-DE-CONDU-2", title: "Conductores y licencias", category: "licencias" },
  { key: "pruebas_teoricas_practicas", guid: "PRUEB-TEORI-Y-PRACT", title: "Pruebas teóricas y prácticas", category: "pruebas" },
  { key: "tabla_interactiva_fallecidos", guid: "DATOS-PARA-TABLA-INTER-94312", title: "Datos para tabla interactiva de fallecidos", category: "fallecidos" },
  { key: "tabla_interactiva_accidentes", guid: "DATOS-PARA-TABLA-INTER-DE", title: "Datos para tabla interactiva de accidentes", category: "accidentes" },
  { key: "accidentes_con_victimas", guid: "ACCID", title: "Accidentes con víctimas", category: "accidentes" },
  { key: "caracteristicas_infracciones", guid: "CARAC-DE-INFRA", title: "Características de infracciones", category: "infracciones" },
  { key: "consulta_infracciones_articulo", guid: "CONSU-DE-INFRA-POR-ARTIC", title: "Consulta de infracciones por artículo", category: "infracciones" },
  { key: "infracciones_detalle", guid: "INFRA-46061", title: "Infracciones detalle", category: "infracciones" },
];

// -----------------------------------------------------------------------------
// Lookup helpers
// -----------------------------------------------------------------------------

/**
 * Returns all known dashboards, optionally filtered by category.
 * Matches the `category` field exactly (case-sensitive).
 */
export function listKnownDashboards(category?: string): KnownCoseviDashboard[] {
  if (!category) return KNOWN_COSEVI_DASHBOARDS;
  return KNOWN_COSEVI_DASHBOARDS.filter(d => d.category === category);
}

/**
 * Looks up a known dashboard by its stable key or its Junar GUID.
 * Returns `undefined` if no match is found.
 */
export function getKnownDashboard(keyOrGuid: string): KnownCoseviDashboard | undefined {
  return KNOWN_COSEVI_DASHBOARDS.find(d => d.key === keyOrGuid || d.guid === keyOrGuid);
}
